### ISSUE TITLE
Hot Shelter waterfall UVs

### ISSUE THREAD

**Issue by PiKeyAr** [SonicFreak94/sadx-fixed-edition#8]
![d598984e-3375-11e7-9c9b-04b21e09257b 1](https://cloud.githubusercontent.com/assets/23312549/26030247/774bdc16-3888-11e7-85cb-ec18f21b8959.jpg)
Display function: sub_5AD3B0
Main function: sub_5AD580
Another function that does something with its UVs: sub_5AD920

This thing is supposed to have its V coordinates shifted while the water fills up the tanks, but in SADX the bottom part of it isn't animated at all. There are supposed to be two arrays of 56 and 20 UV coordinates. The model's meshset order (2 meshes) was swapped in SADX, which is possibly why this happens, though I'm not entirely sure. Replacing the model with anything breaks horizontal stretching of the model for some reason, so I couldn't fix it by simply swapping the model.

**Comment by PiKeyAr** [comment]
I fixed it by doing the following:
1) Disable the UV shifts done in code.
```
WriteData((char*)0x005AD8EB, 0x00, 1);
WriteData((char*)0x005AD89B, 0x00, 1);
WriteData((char*)0x005AD9CE, 0x01, 1);
WriteData((char*)0x005ADA7B, 0x01, 1);
WriteData((void*)0x005AD896, 0x90, 3);
WriteData((void*)0x005AD8E6, 0x90, 3);
```
2) Add custom code to shift UVs manually.

```
DataPointer(int, FramerateSetting, 0x0389D7DC);
static int WaterThing_VShift = 0;
DataPointer(float, HotShelterWaterThing, 0x3C72E54);
```

In an OnFrame function

```
if (CurrentLevel == 12 && CurrentAct == 0 && GameState != 16)
		{
		if (HotShelterWaterThing < 65.0f && HotShelterWaterThing > 0.0f)
			{
			WaterThing_VShift = (WaterThing_VShift - 16 * FramerateSetting) % 255;
			for (int i = 0; i < 56; i++)
				{
				uv_014107E0[i].v = uv_014107E0_0[i].v + WaterThing_VShift;
				}
				for (int i = 0; i < 20; i++)
				{
				uv_01410790[i].v = uv_01410790_0[i].v + WaterThing_VShift;
				}
		}
```
There's probably a more elegant solution than this, but hey, at least it works.

**Comment by PiKeyAr** [comment]
Fixed in f18e3bd4ab1b5bf8a7652cd5bdec465317721548
